pytorch_lightning/utilities/imports.py

```diff
@@ -134,9 +134,9 @@ def __repr__(self) -> str:
 
 _APEX_AVAILABLE = _module_available("apex.amp")
 _BAGUA_AVAILABLE = _package_available("bagua")
-_DEEPSPEED_AVAILABLE = _package_available("deepspeed")
-_DEEPSPEED_GREATER_EQUAL_0_5_9 = _DEEPSPEED_AVAILABLE and _compare_version("deepspeed", operator.ge, "0.5.9")
-_DEEPSPEED_GREATER_EQUAL_0_6 = _DEEPSPEED_AVAILABLE and _compare_version("deepspeed", operator.ge, "0.6.0")
+_DEEPSPEED_AVAILABLE = bool(_RequirementAvailable("deepspeed"))
+_DEEPSPEED_GREATER_EQUAL_0_5_9 = _DEEPSPEED_AVAILABLE and bool(_RequirementAvailable("deepspeed>=0.5.9"))
```

Comment:
I wonder if it would not be enough to make it:
`_DEEPSPEED_GREATER_EQUAL_0_5_9 = bool(_RequirementAvailable("deepspeed>=0.5.9"))`

Another thing that bothers me, there was a discussion in the thread about moving imports to places individual use. For example `_DEEPSPEED_GREATER_EQUAL_0_5_9` is used only in https://github.com/PyTorchLightning/pytorch-lightning/blob/90acf909fb0e7bb34aa3a73dda3ea74e705b9230/tests/strategies/test_deepspeed_strategy.py#L49

Maybe we can move this check directly to the file? (same with the others)

Reply:
I agree I will check the test file if it's only required for the test module. Then I will move that accordingly and upated the PR. how does that sound ?

Reply:
I helped out by pushing the change to this PR :)